core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1962,6 +1963,18 @@ protected Type visitCurrentPath(CurrentPath node, StackableAstVisitorContext<Con
             return setExpressionType(node, VARCHAR);
         }
 
+        @Override
+        protected Type visitTrim(Trim node, StackableAstVisitorContext<Context> context)
+        {
+            List<Type> argumentTypes = node.getArguments().stream()
+                    .map(expression -> process(expression, context))
+                    .collect(toImmutableList());
+
+            String functionName = node.getSpecification().getFunctionName();
+            ResolvedFunction function = plannerContext.getMetadata().resolveFunction(session, QualifiedName.of(functionName), fromTypes(argumentTypes));
+            return setExpressionType(node, function.getSignature().getReturnType());
```

Comment:
WRT the return type -- I think we have a problem there. All the functions should return `varchar(x)`, but half of them declares `char(x)` as the returned type. And from the tests I see that the results are incorrect for `char` values:
trim('abc   ') -> 'abc   ' (spaces trimmed and then padded back to the original length)

Reply:
Fixed in #11440